### PR TITLE
Kevin/Noah blog post

### PR DESCRIPTION
**Pull request by sandertyu**
Alright, the blog post is in a good state. I'm pretty sure it's been spell/sanity checked but may want to give it another full read over by other folks. Feel free to adjust the tags.

### CODE REVIEW

**Review by narrator0 — APPROVED** [review]
Looks great! Love it!

File: content/jupyter-initial-galaxy.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+Keepalived runs as a static kubernetes pod on each control-plane node
+and it manages a virtual IP for the kube-apiserver that all of the
+kubernetes worker nodes communicate with. If the control-plane node
+currently holding this Keepalived virtual IP goes down, Keepalived will
+pass the virtual IP on to one of our remaining control-plane nodes so
+that the rest of the kubernetes cluster can still communicate with the
+kube-apiserver.
+
+After Keepalived recieves a kube-apiserver request, HAproxy (also
```
Comment (celine168) [inline content/jupyter-initial-galaxy.rst:132]:
```suggestion
After Keepalived receives a kube-apiserver request, HAproxy (also
```

File: content/jupyter-initial-galaxy.rst [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+running as a static pod on each control-plane node) will load-balance
+the request onto whichever control-plane node is chosen according to a
+`round robin
+algorithm <https://avinetworks.com/glossary/round-robin-load-balancing/>`__.
+If we did not use HAproxy, any kube-apiserver request recieved through
```
Comment (celine168) [inline content/jupyter-initial-galaxy.rst:137]:
```suggestion
If we did not use HAproxy, any kube-apiserver request received through
```

File: content/jupyter-initial-galaxy.rst [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+the Keepalived virtual IP would automatically be routed to that same
+control-plane node, leaving the other control-plane nodes to just sit
+there and do nothing. Load-balancing with HAproxy takes better advantage
+of our available hardware and lowers the kube-apiserver load on any
+given node.
+
+To improve the maintainability of this cluster, we use
+`puppet <https://puppet.com/docs/puppet/6.18/puppet_index.html>`__ to
+setup kubernetes and administrate our entire bare-metal cluster.
+Althought there already exists a `puppet module for boostrapping
```
Comment (celine168) [inline content/jupyter-initial-galaxy.rst:147]:
```suggestion
Although there already exists a `puppet module for bootstrapping
```

File: content/jupyter-initial-galaxy.rst [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+kubernetes <https://github.com/puppetlabs/puppetlabs-kubernetes>`__, our
+high availability setup has specific demands which required a `new
+puppet module <https://github.com/LibreTexts/protogalaxy>`__ to be
+written. The Protogalaxy puppet module boostraps all of the necessary
```
Comment (celine168) [inline content/jupyter-initial-galaxy.rst:151]:
```suggestion
written. The Protogalaxy puppet module bootstraps all of the necessary
```

File: content/jupyter-initial-galaxy.rst [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+there and do nothing. Load-balancing with HAproxy takes better advantage
+of our available hardware and lowers the kube-apiserver load on any
+given node.
+
+To improve the maintainability of this cluster, we use
+`puppet <https://puppet.com/docs/puppet/6.18/puppet_index.html>`__ to
+setup kubernetes and administrate our entire bare-metal cluster.
+Althought there already exists a `puppet module for boostrapping
+kubernetes <https://github.com/puppetlabs/puppetlabs-kubernetes>`__, our
+high availability setup has specific demands which required a `new
```
Comment (celine168) [inline content/jupyter-initial-galaxy.rst:149]: If possible, maybe mention what the specific demands were?

File: content/jupyter-initial-galaxy.rst [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+been migrated over to v3. Our grafana alerts setup has been recrafted so
+that we no longer have to manually reinput our dashboards if the pod
+goes down. We also enabled IPMI interfaces on all the nodes, affording
+us remote adminstration tools in light of Covid-19.
+
+Future Plans
+------------
+
+Our top priorities moving forward are to enhance the executable code
+cell features of LibreText and continue development on the Galaxy
```
Comment (celine168) [inline content/jupyter-initial-galaxy.rst:199]:
```suggestion
cell features of LibreTexts and continue development on the Galaxy
```

File: content/jupyter-initial-galaxy.rst [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+cluster. We would like to fix the current issues with ipywidgets and
+other interactive plotting features currently exhibited by our CKEditor
+Binder plugin. To compliment this, we must bring cell-to-cell
```
Comment (celine168) [inline content/jupyter-initial-galaxy.rst:202]:
```suggestion
Binder plugin. To complement this, we must bring cell-to-cell
```

**Review by celine168 — CHANGES_REQUESTED** [review]
Thanks for writing this guys; this post looks really great. I really loved learning about the work you all did. 
I don't think I had much to add, most of my comments are just nitpicks on spelling/clarification.

Nit: 'Kubernetes', 'Grafana', 'Dockerfile', 'Puppet' should probably be capitalized.

File: content/jupyter-initial-galaxy.rst [hunk 8]
```diff
@@ -0,0 +1,211 @@
+JupyterTeam Progress Summer 2020
+================================
+
+:date: 2020-10-01 00:00:00
```
[the remark this answers is not in the document]
Comment (sandertyu) [reply content/jupyter-initial-galaxy.rst:4]: I wrote it as "October 10th 2020" which was when I started the document. The format is a little odd admittedly, but do you also want the date to be more like 10/13 ?

[the remark this answers is not in the document]
Comment (moorepants) [reply content/jupyter-initial-galaxy.rst:4]: Yes, make it the 13th, we can merge before the 13th is over. You all are up late!!

File: content/jupyter-initial-galaxy.rst (hunk 5 above)
Comment (sandertyu) [reply content/jupyter-initial-galaxy.rst:149]: I think there I just meant that the official module doesn't install HAproxy and Keepalived. We can probably explain more in detail personally cuz I don't wanna write it all out lol

File: content/jupyter-initial-galaxy.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (celine168) [reply content/jupyter-initial-galaxy.rst]: I read it as October 1st so I suggested a date update.

**Pull request merged**